src/App.tsx

```diff
@@ -1,173 +1,106 @@
-import { Loader } from './components/Loader';
+import {
+  NavLink, Navigate, Route, Routes,
+} from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import cn from 'classnames';
 
 import './App.scss';
 
-export const App = () => (
-  <div data-cy="app">
-    <nav
-      data-cy="nav"
-      className="navbar is-fixed-top has-shadow"
-      role="navigation"
-      aria-label="main navigation"
-    >
-      <div className="container">
-        <div className="navbar-brand">
-          <a className="navbar-item" href="#/">
-            Home
-          </a>
-
-          <a
-            className="navbar-item has-background-grey-lighter"
-            href="#/people"
-          >
-            People
-          </a>
-        </div>
-      </div>
-    </nav>
-
-    <main className="section">
-      <div className="container">
-        <h1 className="title">Home Page</h1>
-        <h1 className="title">People Page</h1>
-        <h1 className="title">Page not found</h1>
-
-        <div className="block">
-          <div className="box table-container">
-            <Loader />
-
-            <p data-cy="peopleLoadingError" className="has-text-danger">
-              Something went wrong
-            </p>
-
-            <p data-cy="noPeopleMessage">
-              There are no people on the server
-            </p>
-
-            <table
-              data-cy="peopleTable"
-              className="table is-striped is-hoverable is-narrow is-fullwidth"
+import { Person } from './types';
+import { NotFoundPage } from './components/NotFoundPage';
+import { HomePage } from './components/HomePage';
+import { PeoplePage } from './components/peoplePage';
+import { getPeople } from './api';
+
+export const App = () => {
+  const [peopleList, setPeopleList] = useState<Person[]>([]);
+  const [loadingError, setLoadingError] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  const loadPeople = async () => {
+    setLoading(true);
+    try {
+      const peopleData = await getPeople();
+
+      setPeopleList(peopleData);
+    } catch {
+      setLoadingError(true);
+    } finally {
+      setLoading(false);
+    }
+  };
```

Comment:
You need to load data only when you are on `people` path (when <PeoplePage /> is rendered)